### PR TITLE
Handle --list-tests in dotnet test

### PR DESCRIPTION
**Pull request by mariam-abdulla**

### CODE REVIEW

**Review by MarcoRossignoli — APPROVED** [review]

File: src/Cli/dotnet/commands/dotnet-test/IPC/ObjectFieldIds.cs [hunk 1]
```diff
@@ -9,55 +9,77 @@ namespace Microsoft.DotNet.Tools.Test
 {
     internal static class CommandLineOptionMessagesFieldsId
```
Comment (MarcoRossignoli) [inline src/Cli/dotnet/commands/dotnet-test/IPC/ObjectFieldIds.cs:10]: I was thinking to something more like 

```cs
internal static class CommandLineOptionMessagesFieldsId
{
     public const int MessagesSerializerId = 3;
     .....
```

so all the info are in the same place for the same object type

File: src/Cli/dotnet/commands/dotnet-test/IPC/ObjectFieldIds.cs [hunk 2]
```diff
@@ -9,55 +9,77 @@ namespace Microsoft.DotNet.Tools.Test
...
+        internal const ushort CommandLineOptionMessageList = 2;
     }
 
     internal static class CommandLineOptionMessageFieldsId
     {
-        internal const int Name = 1;
-        internal const int Description = 2;
-        internal const int IsHidden = 3;
-        internal const int IsBuiltIn = 4;
+        internal const ushort Name = 1;
```
Comment (MarcoRossignoli) [inline src/Cli/dotnet/commands/dotnet-test/IPC/ObjectFieldIds.cs:18]: here and in the other place you don't need to be internal, the class is already internal, public is good

**Review by MarcoRossignoli — COMMENTED** [review]
One more round LGTM

**Review by MarcoRossignoli — APPROVED** [review]

**Comment by Evangelink** [comment]
Relates to #45927

**Pull request merged**